src/main/scala/uncore/tilelink2/RegisterCrossing.scala

```diff
@@ -147,15 +147,18 @@ object AsyncRWSlaveRegField {
     slave_reset: Bool,
     width:  Int,
     init: Int,
+    name: Option[String] = None,
     master_allow: Bool = Bool(true),
     slave_allow: Bool = Bool(true)
   ): (UInt, RegField) = {
 
     val async_slave_reg = Module(new AsyncResetRegVec(width, init))
+    if (name.isDefined) {async_slave_reg.suggestName(name.get)}
```

Comment:
That was exactly the kind of feedback I was looking for, knew there had to be a more scala-y way.